### PR TITLE
[Xamarin.Android.Build.Tasks] Skip library proguard.txt with disallowed global options

### PR DESCRIPTION
**Pull request by jonathanpeppers**
Fixes/aligns with the behavior introduced in [Android Gradle Plugin 9.0](https://developer.android.com/build/releases/agp-9-0-0-release-notes#behavior-changes):

> From AGP 9.0, Android library and feature module publishing will fail if consumer keep files contain problematic Proguard configurations. Consumer keep files that include global options like `-dontoptimize` or `-dontobfuscate` should only be used in application modules, and can reduce optimization benefits for library users. Android App module compilation will silently ignore any such global options if embedded in a pre-compiled dependency (JAR or AAR).

The same restriction applies to R8 for options that control where build outputs go — `-printmapping`, `-printconfiguration`, `-printusage`, `-printseeds`, `-dump`. When a third-party `.aar` (or NuGet package wrapping one) ships a `proguard.txt` with one of those directives, current `dotnet build` either:

* fails the R8 invocation, or
* writes mapping/usage files to the path the library author chose (silently overwriting the location the app project requested).

## Change

In `R8.cs`, after `ResolveLibraryProjectImports` has extracted each `.aar`'s `proguard.txt`, scan the file (only for items that carry the `OriginalFile` metadata — i.e. library-provided files, not ones we generate or the user added). If any disallowed global option is present:

* the whole file is skipped — it is not passed to R8 with `--pg-conf`, and
* a new `XA4322` warning is emitted naming the source: NuGet package id + version when available, otherwise the `.aar` path.

Example warning:

```
warning XA4322: Skipping library ProGuard configuration file 'obj/Debug/net10.0-android/lp/0/jl/proguard.txt' (from NuGet package 'Acme.SomeLibrary' 1.2.3) because it contains the unsupported global option '-printmapping'. Global ProGuard options are only allowed in application projects.
```

User-authored `<ProguardConfiguration Include="..." />` items in the app project, and the SDK-generated configs (`proguard_xamarin.cfg`, `proguard_project_primary.cfg`, etc.) are left untouched.

## Notes

* The `ProguardConfigurationFiles` parameter on the `R8` task changed from `string[]` to `ITaskItem[]` so we can read the `OriginalFile`/`NuGetPackageId`/`NuGetPackageVersion` metadata that `ResolveLibraryProjectImports` already attaches. No targets needed updating — `@(_ProguardConfiguration)` / `@(ProguardConfiguration)` already flow through as items.
* This is more aggressive than AGP, which silently strips just the offending lines. The simpler "skip whole file + warn" was preferred over rewriting library files to a temp location. The XA4322 doc page describes the workaround (add equivalent `-keep` rules to the app's own `ProguardConfiguration`) for users blocked on a 3rd-party library fix.

## Tests

* New `R8Tests.TryGetDisallowedOption` parametric unit test covers each option, leading whitespace (spaces and tabs), comments, empty lines, and case-insensitive matching.
* Manual end-to-end verification can be done by adding `-printmapping out.txt` to any AAR's `proguard.txt` and observing the `XA4322` warning instead of the build failure / silent mapping override.

### CODE REVIEW

File: src/Xamarin.Android.Build.Tasks/Tasks/R8.cs [hunk 1]
```diff
@@ -155,19 +155,86 @@ protected override string CreateResponseFile ()
...
+		//   https://developer.android.com/build/releases/agp-9-0-0-release-notes#behavior-changes
+		// We skip the whole offending file and emit a warning naming the source library
+		// so the build can still succeed.
+		static readonly string [] DisallowedLibraryProguardOptions = {
+			"-dump",
+			"-printconfiguration",
+			"-printmapping",
+			"-printseeds",
+			"-printusage",
+		};
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply src/Xamarin.Android.Build.Tasks/Tasks/R8.cs:192]: Good catch. Added `-dontoptimize` and `-dontobfuscate` to `DisallowedLibraryProguardOptions` in fbff89d. Those are the primary cases AGP 9.0 calls out, so they belong in the list.

File: src/Xamarin.Android.Build.Tasks/Tasks/R8.cs [hunk 2]
```diff
@@ -155,19 +155,86 @@ protected override string CreateResponseFile ()
...
+			var trimmed = line.TrimStart ();
+			foreach (var candidate in DisallowedLibraryProguardOptions) {
+				if (trimmed.StartsWith (candidate, StringComparison.OrdinalIgnoreCase)) {
+					option = candidate;
+					return true;
+				}
+			}
+			option = "";
+			return false;
+		}
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply src/Xamarin.Android.Build.Tasks/Tasks/R8.cs:227]: Fixed in fbff89d — `TryGetDisallowedOption` now requires an end-of-token boundary (end-of-line or whitespace) after the matched option, so `-printmappingFoo` no longer matches `-printmapping`. Tests cover both `-printmappingFoo` and `-dumpsterfire`.

File: src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/R8Tests.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+		[TestCase ("# -printmapping comment",            false, "")]
+		[TestCase ("",                                   false, "")]
+		[TestCase ("-printmapping mapping.txt",          true,  "-printmapping")]
+		[TestCase ("  -printmapping mapping.txt",        true,  "-printmapping")]
+		[TestCase ("\t-printseeds seeds.txt",            true,  "-printseeds")]
+		[TestCase ("-printusage usage.txt",              true,  "-printusage")]
+		[TestCase ("-printconfiguration config.txt",     true,  "-printconfiguration")]
+		[TestCase ("-dump dump.txt",                     true,  "-dump")]
+		[TestCase ("-PrintMapping mapping.txt",          true,  "-printmapping")] // case-insensitive
+		[TestCase ("-DUMP dump.txt",                     true,  "-dump")]
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/R8Tests.cs:25]: Added in fbff89d: test cases for `-dontoptimize`, `-dontobfuscate` (including case-insensitive `-DontOptimize`), the bare option-with-no-argument case (`-printmapping`), and the boundary cases `-printmappingFoo` and `-dumpsterfire`.

File: Documentation/docs-mobile/messages/xa4322.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+```
+warning XA4322: Skipping library ProGuard configuration file 'obj/Debug/net10.0-android/lp/0/jl/proguard.txt' (from NuGet package 'Acme.SomeLibrary' 1.2.3) because it contains the unsupported global option '-printmapping'. Global ProGuard options are only allowed in application projects.
+```
+
+## Issue
+
+An Android library (`.aar`) shipped a `proguard.txt` file (a "consumer ProGuard
+configuration") that contains a global ProGuard option such as `-printmapping`,
+`-printconfiguration`, `-printusage`, `-printseeds`, or `-dump`.
+
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply Documentation/docs-mobile/messages/xa4322.md:23]: Updated in fbff89d — the doc page now lists `-dontoptimize` and `-dontobfuscate` alongside the `-print*` / `-dump` directives.

**Review by simonrozsival — APPROVED** [review]
